tahoe_gui/tahoe.py

```diff
@@ -15,48 +15,39 @@
     os.environ["PATH"] += os.pathsep + os.path.join(
         os.path.dirname(sys.executable), 'Tahoe-LAFS')
 
+class TahoeEnvironment:
+    """
+    I represent a virtualenv that has Tahoe-LAFS installed in it.
+    """
+    def __init__(self, basedir):
+        self._base = basedir
 
-class CommandProtocol(ProcessProtocol):
-    def __init__(self, parent, callback_trigger=None):
-        self.parent = parent
-        self.trigger = callback_trigger
-        self.done = Deferred()
-
-    def outReceived(self, data):
-        for line in data.decode('utf-8').strip().split('\n'):
-            self.parent.out_received(line)
-            if not self.done.called and self.trigger and self.trigger in line:
-                self.done.callback(None)
-
-    def errReceived(self, data):
-        for line in data.decode('utf-8').strip().split('\n'):
-            self.parent.err_received(line)
-
-    def processEnded(self, reason):
-        if not self.done.called:
-            self.done.callback(None)
-
-    def processExited(self, reason):
-        if not self.done.called and not isinstance(reason.value, ProcessDone):
-            self.done.errback(reason)
+    @inlineCallbacks
+    def create(self):
+        """
+        :return: a Deferred that fires when we've installed a virtualenv
+        """
 
 
-class Tahoe():
-    def __init__(self, nodedir=None):
-        self.nodedir = nodedir
+class TahoeClient:
+    def __init__(self, nodedir):
```

Comment:
The `Tahoe` (or `TahoeClient`, as you've rightly renamed it) class was intended to provide a higher-level abstraction layer, each object-instance of which (optionally) corresponds to a particular nodedir and which provides convenience methods to do `tahoe`-related things (like starting/stopping gateways) in a unified way despite the fact that those ways might differ across platforms under the hood (i.e., Windows).. The `Core` object (of which there should only be one per running GUI instance) instantiates, starts, and stops these (which, in turn, starts and stops our Tahoe-LAFS gateways when the GUI runs/exits).

I'm still not entirely sure what you have in mind for `TahoeEnvironment` here (especially given our recent discussion in Issue #3!) -- though I have had a pretty long day and am presently exhausted.. 

Nevertheless, my thinking is that we probably should strive to avoid having to manage a virtualenv if we can and just encourage GNU/Linux users to `apt-get install tahoe-lafs` (or whatever) before installing tahoe-gui (since, presumably, they'd be installing tahoe-gui it from the command line anyway until we start creating Debian/Redhat/etc. packages). Also, if I'm not mistaken, `pip` alone isn't actually sufficient to install tahoe-lafs on a fresh Debian install (since the user will still need packages like python-dev, libssl-dev, libffi-dev, etc. -- unless that's changed). I'll have to double-check this, however, as it's been a while since I've seen/used a fresh install. :)